### DISCUSSION TITLE
Doorontwikkeling ZGW API's: Behoeften en prioriteiten

### DISCUSSION THREAD

**Discussion by michielverhoef** [VNG-Realisatie/gemma-zaken#2326] [Polls]
In de laatste release zijn enkele belangrijke knelpunten opgelost, te weten de problemen rond performance en documentflow. Dit onderwerp wordt nog verder uitgewerkt maar we willen de API’s natuurlijk nog verder verbeteren. Omdat we wel de dingen willen doen waar behoefte aan is hebben we ook hier een enquête gemaakt van een aantal onderwerpen die uit de bestaande issues opgehaald zijn. Dit is een peiling op thema, bijvoorbeeld:

- Document flow
- Verbeteren schrijven en transacties
- Notificaties
- Autorisaties
- Bevragen over meerdere registers/bronnen
- ...

NB. Op sommige onderwerpen wordt al samen met stakeholders gewerkt aan verbetering maar we willen graag een beeld van de totale behoeften. Het kan goed zijn dat wanneer er behoefte aan is meerdere onderwerpen in één release worden opgepakt.

**Comment by EduardWitteveen** [comment]
Ook dat het de standaard gestaag wordt doorontwikkeld

**Comment by johannesbattjes** [comment]
Vanuit Visma Roxit hebben wij de volgende speerpunten:

Dringende documentfunctionaliteit (hebben we al samen met klanten en Michiel Verhoef besproken):
https://github.com/VNG-Realisatie/gemma-zaken/issues/2242
https://github.com/VNG-Realisatie/gemma-zaken/issues/2079

Langlopende oneffenheden in de formulering van de standaard die testen en samenwerking met andere partijen sterk bemoeilijken:
https://github.com/VNG-Realisatie/gemma-zaken/issues/1878
https://github.com/VNG-Realisatie/gemma-zaken/issues/2067
https://github.com/VNG-Realisatie/gemma-zaken/issues/2090
https://github.com/VNG-Realisatie/gemma-zaken/issues/1901
 
Ondersteuning wettelijke geo-standaarden:
https://github.com/VNG-Realisatie/gemma-zaken/issues/2277

Lang openstaande archiveringsproblemen:
https://github.com/VNG-Realisatie/gemma-zaken/issues/1929 
https://github.com/VNG-Realisatie/gemma-zaken/issues/2086

Daarnaast heeft Decos al eerder gemeld dat de referentie naar betrokkenen vanuir rol niet goed geregeld is. Dit levert veel praktijkproblemen en moet ook worden opgelost.

**Reply by davidbronsveld** [reply]
voor wat betreft laatste punt een korte aanvulling: wij hebben nu een StUFBG brug ontwikkeld (in overleg met Michiel) om in ieder geval een betrokkene (NP) te kunnen koppelen. We zijn momenteel ook bezig om NNP op basis van RSIN te kunnen koppelen. hierbij lopen wij aan tegen het probleem dat bepaalde vereiste waarden zoals BSN en RSIN niet verplicht zijn en daarmee tot fouten kunnen leiden in het koppelvlak.

**Comment by joeribekker** [comment]
Notificaties en Autorisaties zouden echt hun eigen projecten moeten worden, met een apart ontwikkeltraject. Dit maakt ze ook breder toepasbaar (en met input van) van bijvoorbeeld Haal Centraal. Binnen het ZGW traject zou ik hier niet de focus op leggen.

Ik zou graag zien dat de ZGW APIs (zaken, documenten, catalogi en besluiten) gestaag door worden ontwikkeld zoals dat de laatste maanden gaat, met periodieke kleinere releases. Wellicht weer wat meetings :) 

Als Documenten helemaal zelfstandig kunnen bestaan is dat ook erg fijn (dus met eigen archiefparameters en wellicht niet verplicht gekoppeld aan zaken (dus ook niet aan de ZAAKtypecatalogus).

Laat ik ook expliciet zijn over transacties en schrijven: Dit lijkt me OF een veel groter topic dan ZGW OF het kan gewoon prima opgelost worden door leveranciers zonder dat het de standaard beïnvloedt. Bevragen over meerdere bronnen geldt eigenlijk hetzelfde voor.

**Comment by johannesbattjes** [comment]
Onder hoofdstukje oneffenheden ben ik deze vergeten:
https://github.com/VNG-Realisatie/gemma-zaken/issues/1812

dit is echt heel onhandig bij testen we raken hier in overleg met andere leveranciers heel veel tijd mee kwijt
